packages/hardhat-core/test/internal/hardhat-network/provider/node.ts

```diff
@@ -601,6 +601,54 @@ describe("HardhatNode", () => {
         await assertIncreaseTime(now + delta + elapsedTimeInSeconds);
       });
 
+      it("mines blocks with the same timestamp if allowBlocksWithSameTimestamp is set", async () => {
+        [, node] = await HardhatNode.create({
+          ...config,
+          allowBlocksWithSameTimestamp: true,
+        });
+
+        const timestamps = [];
+
+        for (let i = 0; i < 10; i++) {
+          await node.mineBlock();
+          timestamps.push((await node.getLatestBlock()).header.timestamp);
+        }
+
+        let differentTimestamps = 0;
+
+        for (let i = 0; i + 1 < timestamps.length; i++) {
+          if (timestamps[i] !== timestamps[i + 1]) {
+            differentTimestamps++;
+          }
+        }
+
+        // There is a small chance that two of the mined blocks were mined in a
+        // different second.
+        // This number shouldn't be bigger than 1 if we are running these tests
+        // on a computer from this century.
```

Comment:
If we expect hardhat to be here for the long run we should clarify which century is `this century`.